envoy/http/filter.h

```diff
@@ -1043,6 +1043,12 @@ class FilterChainFactoryCallbacks {
    * @param handler supplies the handler to add.
    */
   virtual void addAccessLogHandler(AccessLog::InstanceSharedPtr handler) PURE;
+
+  /**
+   * Allows filters to access the thread local dispatcher on creation.
```

Comment:
Not just creation, any other time as well right? If we want to limit this to only being permitted at creation we could add this to the FactoryContext (might be hard?) or add a parameter to the HTTP filter factory (annoying refactor probably).

Is there any reason we'd want to prevent general access to this? If no then I think what this PR does is fine, modulo the comment

Reply:
sorry the comment is unclear.  Filters already had access to the dispatcher once they got their set[encoder][decoder]callbacks function, but they didn't have access until that point.  now they have access as early as creation.
Will update the comment for clarity.

Reply:
Oh okay, yeah this makes sense to me then

Reply:
cool.  tests are added and first review is passed so do take a look overall when you have a few!